### DISCUSSION TITLE
[BATTERY LIMIT] TASK run "AT LOGON" versus "AT STARTUP"

### DISCUSSION THREAD

**Discussion by SFX72** [seerge/g-helper#1773] [General]
Hi,

My ASUS TUF FX567ZC4 (like FX507ZC4) is charging during the Windows (10 pro) startup 
although the limitation set at 60% for my case. The battery charge value gradually (slowly) increases 
every time Windows starts.

G-Helper (checked "run on startup") is launching just after (short time) the user logon process 
in accordance with the dedicated Task created in 'Task scheduler' with trigger option 'At logon'.

Wouldn't it be interesting to create a Task scheduler with trigger option really 'At Startup' ?
I tried manually to create this task but didn't succeed.

It doesn't really have any impact just the value increases gradually.

Does anyone have any ideas on this question ?

**Comment by seerge** [comment]
@SFX72 hello, you can't start G-Helper at startup as then it will run w/o user and you won't be able to access app's UI (forms), check other discussion on the same topic https://github.com/seerge/g-helper/discussions/1726 and my advice there

**Reply by SFX72** [reply]
Hi Seerge,  Thank you for your answer and the reference on the same subject #1726 which I had not seen. 
yes i noticied there is no access UI with 'startup' task, it was expected :( 
Also i tried to apply (cohabit) 2 tasks the 1st with 'Startup' trigger and the 2nd created by G-Helper but didn't succeed.
Maybe the 2nd (with UI) would kill the 1st task if it is present . . . 
thanks again

**Reply by seerge** [reply]
@SFX72 currently app is designed in a way, that when you start a new GHelper.exe it will kill all existing copies of itself

You may experiment by creating 2 tasks, 1 to run on startup, 1 at logon (default), both as admin. Mb it will help you :)

**Reply by SFX72** [reply]
@seerge Thanks for the advice, I'll try again. Sometimes 'Task scheduler' is not very intuitive :)

**Reply by SFX72** [reply]
Hi,
I tried with 2 planned tasks but it doesn't work. 
Maybe there will be a fix in the next BIOS version.
Happy New Year

**Comment by AudreyAP** [comment]
Try playing with the fast startup option in BIOS (not windows fast startup). on some machines, enabling it or disabling it can stop the charge limit from resetting on reboot. It varies wildly by model.

**Reply by SFX72** [reply]
Hi AudreyAP, 

thanks for the method (with and without fast startup in BIOS) but there is no change with my model.

Happy New Year

**Comment by MC-dusk** [comment]
Check your win11 (if it is win11, win10 or before idk) setting, `account > login option > other options > use my login info to auto finish setting after update`, which is before `related settings`. That's translated by me, idk what it is in english system.

Also for #1726

**Reply by seerge** [reply]
@MC-dusk nice find! this actually works ... i.e. app starts even before I enter my PIN / pass face id

![Screenshot 2024-02-15 163037](https://github.com/seerge/g-helper/assets/5920850/20db04bf-9df8-40df-8363-490883639341)

**Reply by seerge** [reply]
I have updated FAQ page with this screenshot and info. Thanks

**Reply by Kikimahe** [reply]
Hi @MC-dusk and @seerge, 

Sorry, but I don't understand how this setting can improve the startup of GHelper. :'( 
- This setting is already ON for me.
- I also removed the password & pin code prompt at startup.

However, Ghelper only starts when the Windows desktop is displayed.

**Reply by seerge** [reply]
@Kikimahe well, for me windows seem to start it before that (at least when I reboot / shutdown w/o changing  user)

I literally see how charge led becomes yellow (charging) after reboot - and then as soon as login screen appears - it becomes white (g-helper starts and sets limit)

**Reply by Kikimahe** [reply]
OK I understand.

I'm just looking at the Ghelper icon in the taskbar.
It is true. It's seems that the charging light turns off long before the Ghelper icon appears.

Thanks Seerge. 👍

**Reply by seerge** [reply]
@Kikimahe cause G-Helper is Winforms app (i.e. app with a window), it can "officially" be launched only from some user. But apparently that setting in windows bypasses it somehow, and lets app start earlier. And UI appears only when actual user logs in.
